md-dao/src/main/java/com/wipro/ats/bdre/md/dao/HistoryDataImportDAO.java

```diff
@@ -696,7 +690,7 @@
                     rawToStage.setNextProcessId(nextProcessForR2S);
                     session.update(rawToStage);
                 }
-                if (parentProcessIdList.size() != 0) {
+                if (!parentProcessIdList.isEmpty()) {
                     Criteria parentProcessCriteria = session.createCriteria(Process.class).add(Restrictions.in("processId", parentProcessIdList));
                     LOGGER.info("size of parent process list is" + parentProcessCriteria.list().size());
                     for (Object resultObject : parentProcessCriteria.list()) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)